src/WinRT.Runtime/Marshalers.cs

```diff
@@ -54,6 +56,7 @@ public static unsafe MarshalString CreateMarshaler(string value)
             try
             {
                 m._gchandle = GCHandle.Alloc(value, GCHandleType.Pinned);
+                m._gchandleHeader = GCHandle.Alloc(m._header, GCHandleType.Pinned);

```

Comment:
I think this will end up making a boxed copy of the header and pinning that.

Reply:
D'oh. You could be right. I will double check it and attempt a different fix if that turns out to be the case.

Reply:
Hopefully fixed by pinning the `MarshalString` object itself. I'm widely uncomfortable with the layout of that class. Just want to have something working with minimal changes before trying to improve on the fix.

Reply:
Pinning the `MarshalString` object itself should work since it's a class.